dask/dataframe/tests/test_groupby.py

```diff
@@ -553,3 +557,71 @@ def test_groupby_multiprocessing():
     with dask.set_options(get=get):
         assert_eq(ddf.groupby('B').apply(lambda x: x),
                   df.groupby('B').apply(lambda x: x))
+
+
+@pytest.mark.parametrize('spec', [
+    {'b': {'c': 'mean'}, 'c': {'a': 'max', 'a': 'min'}},
+    {'b': 'mean', 'c': ['min', 'max']},
+    {'b': np.sum, 'c': ['min', np.max, np.std, np.var]},
+    ['sum', 'mean', 'min', 'max', 'count', 'size', 'std', 'var'],
+    'var',
+])
+@pytest.mark.parametrize('split_every', [False, 3, None])
+@pytest.mark.parametrize('grouper', [
+    lambda df: 'a',
+    lambda df: ['a', 'd'],
+    lambda df: [df['a'], df['b']],
+])
+def test_aggregate__examples(spec, split_every, grouper):
+    pdf = pd.DataFrame({'a': [1, 2, 3, 1, 1, 2, 4, 3, 7] * 20,
+                        'b': [4, 2, 7, 3, 3, 1, 1, 1, 2] * 20,
+                        'c': [0, 1, 2, 3, 4, 5, 6, 7, 8] * 20,
+                        'd': [3, 2, 1, 3, 2, 1, 2, 6, 4] * 20},
+                       columns=['c', 'b', 'a', 'd'])
+    ddf = dd.from_pandas(pdf, npartitions=20)
+
+    assert_eq(pdf.groupby(grouper(pdf)).agg(spec),
+              ddf.groupby(grouper(ddf)).agg(spec, split_every=split_every))
+
+
+@pytest.mark.parametrize('spec', [
+    'sum', 'mean', 'min', 'max', 'count', 'size', 'std', 'var'
+])
+def test_aggregate__single_element_groups(spec):
+    pdf = pd.DataFrame({'a': [1, 2, 3],
+                        'b': [4, 2, 7],
+                        'c': [0, 1, 2],
+                        'd': [1, 2, 3]},
+                       columns=['c', 'b', 'a', 'd'])
+    ddf = dd.from_pandas(pdf, npartitions=3)
+
+    assert_eq(pdf.groupby(['a', 'd']).agg(spec),
+              ddf.groupby(['a', 'd']).agg(spec))
+
```

Comment:
After adapting the tests. I found a couple of problems:

1. using series as a groupby argument fails miserably
2. pandas is using additional magic where it tries to cast the result back to its original type. The relevant code can be found in [_GroupBy._try_cast](https://github.com/pandas-dev/pandas/blob/master/pandas/core/groupby.py#L760). (see also #1685). While it is definitely possible to ensure the final data frame has the correct types after `.compute()`. This would also mean, that there is no way of knowing the correct meta object before performing the actual aggregation
3. `.groupby(...).agg('size')` returns a series for inconsistence sake